chaste_codegen/chaste_model.py

```diff
@@ -350,103 +318,151 @@ def _get_modifiable_parameters(self):
+        negate_stimulus = False
+        voltage_rhs = deq.rhs
+        variables = list(voltage_rhs.free_symbols)
+        for variable in variables:
+            if self._membrane_stimulus_current_orig != variable:
+                if self._membrane_stimulus_current_orig.units.dimensionality == variable.units.dimensionality:
+                    if isinstance(voltage_rhs, sp.expr.Expr):
+                        voltage_rhs = voltage_rhs.xreplace({variable: 0.0})  # other currents = 0
+                else:
+                    # For other variables see if we need to follow their definitions first
+                    rhs = None
+                    if variable in [eq.lhs for eq in d_eqs]:
+                        rhs = [eq.rhs for eq in d_eqs if eq.lhs == variable][-1]
+
+                    if rhs is not None and not isinstance(rhs, sp.numbers.Float):
+                        voltage_rhs = voltage_rhs.xreplace({variable: rhs})  # Update definition
+                        variables.extend(rhs.free_symbols)
+                    else:
+                        if isinstance(voltage_rhs, sp.expr.Expr):
+                            voltage_rhs = voltage_rhs.xreplace({variable: 1.0})  # other variables = 1
+        if isinstance(voltage_rhs, sp.expr.Expr):
+            voltage_rhs = voltage_rhs.xreplace({self._membrane_stimulus_current_orig: 1.0})  # stimulus = 1
+            # plug in math functions for sign calculation
+            voltage_rhs = voltage_rhs.subs(MATH_FUNC_SYMPY_MAPPING)
+        negate_stimulus = voltage_rhs > 0.0
+
+        membrane_stimulus_current_converted = self._model.convert_variable(self._membrane_stimulus_current_orig,
+                                                                           self.uA_per_cm2, DataDirectionFlow.INPUT)
+
+        # Replace stim = ... with stim = +/-GetIntracellularAreaStimulus(t)
+        GetIntracellularAreaStimulus = sp.Function('GetIntracellularAreaStimulus', real=True)(self._time_variable)
+        if negate_stimulus:
+            GetIntracellularAreaStimulus = -GetIntracellularAreaStimulus
+
+        # Get stimulus defining equation
+        eq = self._model.get_definition(membrane_stimulus_current_converted)
+        self._model.remove_equation(eq)
+        self._model.add_equation(sp.Eq(membrane_stimulus_current_converted, GetIntracellularAreaStimulus))
+
+        # Annotate the converted stimulus current as derived quantity
+        self._set_is_metadata(membrane_stimulus_current_converted, 'derived-quantity')
+        return membrane_stimulus_current_converted, \
+            (self._membrane_stimulus_current_orig.units, membrane_stimulus_current_converted.units)
+
     def _get_membrane_capacitance(self):
-        """ Find membrane_capacitance if the model has it"""
-        equation, capacitance_factor = None, 1.0
-        # add membrane_capacitance if the model has it
+        """ Find membrane_capacitance if the model has it and convert it to uA/uA_per_cm2 if necessary"""
```

Comment:
The units here look wrong!

Reply:
You're right of course, I changed it to converting the membrane_capaticatnce to uF and using a only a conversion rule to convert amplitude & ionic currents